### PR TITLE
Update rbac.md

### PR DESCRIPTION
**Pull request by Abirdcfly**

### CODE REVIEW

File: content/en/docs/reference/access-authn-authz/rbac.md [hunk 1]
```diff
@@ -279,7 +279,7 @@ rules:
 ```
 
 {{< note >}}
-You cannot restrict `create` or `deletecollection` requests by resourceName. For `create`, this
+You cannot restrict `create` or `deletecollection` or `list` requests by resourceName. For `create`, this
 limitation is because the object name is not known at authorization time.
 {{< /note >}}
```
Comment (sftim) [inline content/en/docs/reference/access-authn-authz/rbac.md:284]: Ah yes. This has caught me out before!

I think I'd change upgrade this to a caution callout:
```suggestion
{{< caution >}}
You cannot restrict `create` or `deletecollection` or `list` requests by their resource name.

If you specify `resourceNames` along with the `list` verb, you are granting access to `get` all the resources
and not just the resource you named. This is because `list` returns a collection of all the matching resources.

For `create`, this limitation is because the name of the new object may not be known at authorization time.
{{< /caution >}}
```

**Review by sftim — COMMENTED** [review]
LGTM

/sig auth

**Comment by sftim** [comment]
@Abirdcfly: to help reviewers, could you summarize this change in the PR title and also in the PR description?

File: content/en/docs/reference/access-authn-authz/rbac.md [hunk 2]
```diff
@@ -279,7 +279,7 @@ rules:
 ```
 
 {{< note >}}
-You cannot restrict `create` or `deletecollection` requests by resourceName. For `create`, this
+You cannot restrict `create` or `deletecollection` or `list` requests by resourceName. For `create`, this
```
Comment (liggitt) [inline content/en/docs/reference/access-authn-authz/rbac.md:282]: list and watch actually _can_ be restricted by resourceName, and permits "lists" requests using a fieldSelector to "list" a single object (for example, `/api/v1/namespaces/$ns/configmaps?fieldSelector=metadata.name=foo`)

This is useful when using an informer to list/watch a single object

Comment (sftim) [reply content/en/docs/reference/access-authn-authz/rbac.md:282]: Today I learned!
What might help, then, is to document that if you restrict `list` by `resourceName`, the user / client _must_ query using that field selector (is that right?)

Comment (liggitt) [reply content/en/docs/reference/access-authn-authz/rbac.md:282]: yup

**Review by sftim — DISMISSED** [review]
@Abirdcfly based on @liggitt's feedback, we'll need to revise this to make it helpful. The current document is already accurate.

File: content/en/docs/reference/access-authn-authz/rbac.md (hunk 2 above)
Comment (Abirdcfly) [reply content/en/docs/reference/access-authn-authz/rbac.md:282]: According to the test I mentioned above, it is possible to use list for a specific resource. But how to use watch?

Comment (liggitt) [reply content/en/docs/reference/access-authn-authz/rbac.md:282]: including the `watch` verb in the role would allow watching a specific resource by name using a fieldSelector

**Comment by Abirdcfly** [comment]
Thanks for both of you! I learned a lot today! I'll follow your suggestions to update the docs. 
Some tests have been added above so that those who see this issue can get some additional information

File: content/en/docs/reference/access-authn-authz/rbac.md (hunk 1 above)
Comment (Abirdcfly) [reply content/en/docs/reference/access-authn-authz/rbac.md]: change note to caution

File: content/en/docs/reference/access-authn-authz/rbac.md (hunk 2 above)
Comment (Abirdcfly) [reply content/en/docs/reference/access-authn-authz/rbac.md]: Forgive me for the mistake.😂 The documentation has been updated

File: content/en/docs/reference/access-authn-authz/rbac.md [hunk 3]
```diff
@@ -278,10 +278,11 @@ rules:
...
 
-{{< note >}}
-You cannot restrict `create` or `deletecollection` requests by resourceName. For `create`, this
-limitation is because the object name is not known at authorization time.
-{{< /note >}}
+{{< caution >}}
+You cannot restrict `create` or `deletecollection` requests by their resource name.
+For `create`, this limitation is because the name of the new object may not be known at authorization time.
+`list`(and `watch`)actually can be restricted by their resource name, and permits list(and `watch`) requests using a fieldSelector of `metadata.name=...` to match a single object (for example, `/api/v1/namespaces/$ns/configmaps?fieldSelector=metadata.name=foo`). This is useful when using an informer to list/watch a single object.
+{{< /caution >}}
```
Comment (sftim) [inline content/en/docs/reference/access-authn-authz/rbac.md:285]: - I got it wrong. We don't need a `caution` because there isn't a snag like I thought. `note` was right.

- Not all readers will recognize “informer”, which is a client-side helper component. Clients in Golang often use informers and _some_ other clients have similar mechanisms, but not all. I would not mention informer here.

The detail to mention is that:
- if you restrict `list` and `watch` by `resourceName`, then the _only_ way that a client including (`kubectl`) can perform that `list` or `watch` is by specifying a field selector that matches on `metadata.name`.

Comment (Abirdcfly) [reply content/en/docs/reference/access-authn-authz/rbac.md]: Update.

File: content/en/docs/reference/access-authn-authz/rbac.md [hunk 4]
```diff
@@ -279,8 +279,9 @@ rules:
 ```
 
 {{< note >}}
-You cannot restrict `create` or `deletecollection` requests by resourceName. For `create`, this
-limitation is because the object name is not known at authorization time.
+You cannot restrict `create` or `deletecollection` requests by their resource name.
+For `create`, this limitation is because the name of the new object may not be known at authorization time.
+If you restrict `list` or `watch` by resourceName, then the only way that a client including kubectl can perform that `list` or `watch` is by specifying a field selector that matches on metadata.name.
```
Comment (liggitt) [inline content/en/docs/reference/access-authn-authz/rbac.md:284]: Technical accuracy lgtm. Rephrasing the kubectl portion as an example might be clearer, but I'll defer to the docs reviewer:

```suggestion
If you restrict `list` or `watch` by resourceName, clients must include a `metadata.name` field selector in their `list` or `watch` request that matches the specified resourceName in order to be authorized.
For example, `kubectl get configmaps --field-selector=metadata.name=my-configmap`
```

**Comment by sftim** [comment]
LGTMed in https://github.com/kubernetes/website/pull/29468#discussion_r693925022

/lgtm
/approve

**Comment by sftim** [comment]
Thanks @Abirdcfly @liggitt
